lib/repl.js

```diff
@@ -86,7 +99,10 @@ function REPLServer(prompt, stream, eval_, useGlobal, ignoreUndefined) {
 
   self.useGlobal = !!useGlobal;
   self.ignoreUndefined = !!ignoreUndefined;
-
+  self.useHistory = !!useHistory;
+  if (self.useHistory) {
+    self.historyPath = process.env['NODE_HISTORY_PATH'] || historyPath;
```

Comment:
This could be dangerous with setuid binaries.  (Hopefully, no one does that but...)